### PR TITLE
Add Cow helpers for IString

### PR DESCRIPTION
**Pull request by hamza1311**
This PR adds:
- `IString::as_cow`
- `impl From<Cow<'static, str>> for IString`

### CODE REVIEW

**Comment by hamza1311** [comment]
The Clippy failure is unrelated. It's likely that the derived implementation of PartialEq that clippy is highlighting is wrong

File: src/string.rs [hunk 1]
```diff
@@ -164,4 +189,16 @@ mod test_string {
...
+    #[test]
+    fn as_cow_does_not_clone() {
+        let rc_s = Rc::from("foo");
+        let s = IString::Rc(Rc::clone(&rc_s));
+        assert_eq!(Rc::strong_count(&rc_s), 2);
+        let cow: Cow<'_, str> = s.as_cow();
+        assert_eq!(Rc::strong_count(&rc_s), 2);
+        // this assert exists to ensure the cow lives after the strong_count asset
+        assert_eq!(cow, "foo");
+
```
Comment (cecton) [inline src/string.rs:202]:
```suggestion
```

File: src/string.rs [hunk 2]
```diff
@@ -65,6 +81,15 @@ impl From<Rc<str>> for IString {
...
 }
 
+impl From<Cow<'static, str>> for IString {
+    fn from(cow: Cow<'static, str>) -> Self {
+        match cow {
+            Cow::Borrowed(s) => IString::Static(s),
+            Cow::Owned(s) => s.into(),
+        }
+    }
+}
```
Comment (cecton) [inline src/string.rs:91]: I think you forgot to write a test for this. It's not super important but it makes sure that someone does not remove it by mistake

**Review by cecton — APPROVED** [review]
I'm just seeing this now but it's nice! You should have ping me.

You can disable the lint for the impl of Eq at crate level I think it's not relevant here. We don't want to enforce the user to implement Eq on all their types otherwise they won't be able to store `f32` and `f64` in an IArray

**Comment by hamza1311** [comment]
I'm going to merge this. The test can be added later

**Pull request merged**